lib/resources/pg_ident_conf.rb

```diff
@@ -0,0 +1,88 @@
+# encoding: utf-8
+# copyright: 2017
+# author: Rony Xavier,  rx294@nyu.edu
+# author: Aaron Lippold, lippold@gmail.com
+# license: All rights reserved
+
+require 'resources/postgres'
+
+module Inspec::Resources
+  class PGIdentConf < Inspec.resource(1)
+    name 'pg_ident_conf'
+    desc 'Use the pg_ident_conf InSpec audit resource to test the client
+          authentication data is controlled by a pg_ident.conf file.'
+    example "
+      describe pg_ident_conf.where { pg_username == 'acme_user' } do
+        its('map_name') { should eq ['ssl-test'] }
+      end
+    "
+
+    attr_reader :params, :conf_dir, :conf_file
+
+    def initialize(ident_conf_path = nil)
+      @conf_dir = inspec.postgres.conf_dir
+      @conf_file = ident_conf_path || File.expand_path('pg_ident.conf', inspec.postgres.conf_dir)
+      @files_contents = {}
+      @content = nil
+      @params = nil
+      read_content
+      return skip_resource '`pg_ident_conf` is not yet supported on your OS' if inspec.os.windows?
+    end
+
+    def content
+      @content ||= read_content
+    end
+
+    filter = FilterTable.create
+    filter.add_accessor(:where)
+          .add_accessor(:entries)
+          .add(:map_name,        field: 'map_name')
+          .add(:system_username, field: 'system_username')
+          .add(:pg_username,     field: 'pg_username')
+
+    filter.connect(self, :params)
+
+    def to_s
+      "PG Ident Config #{@conf_file}"
+    end
+
+    private
+
+    def filter_comments(data)
+      content = []
+      data.each do |line|
+        if !line.match(/^\s*#/) && !line.chomp.empty?
```

Comment:
You probably want to chomp the line anyways before appending it to your `content` array... otherwise, when you process the lines, they'll all have newlines in the last parameter.

Consider rewriting this to:

```ruby
data.each do |line|
  line.chomp!
  content << line unless line.match(/^\s*#/)
end
```